### ISSUE TITLE
Change behavior when repeating last move from builtin_f_expr and  builtin_t_expr

### ISSUE THREAD

**Issue by gabs712** [nvim-treesitter/nvim-treesitter-textobjects#699]
**Is your feature request related to a problem? Please describe.**
When using `repeat_last_move`on operation pending mode, after triggering`builtin_f_expr`
or `builtin_t_expr`, it operates on one character less that it would be expected to.

Taking this snippet as an exemple:
```sh
vim.keymap.set({ 'n', 'x', 'o' }, ';', ts_repeat_move.repeat_last_move)
```
If the user presses `f(`, returns to the first character and presses `d;`, it would result on:
```sh
({ 'n', 'x', 'o' }, ';', ts_repeat_move.repeat_last_move)
```
`t(_d;` would result on:
```sh
t({ 'n', 'x', 'o' }, ';', ts_repeat_move.repeat_last_move)
```

**Describe the solution you'd like**
For `f(_d;`, it would be expected to behave, as:
```sh
{ 'n', 'x', 'o' }, ';', ts_repeat_move.repeat_last_move)
```
and for `t(_d;`:
```sh
({ 'n', 'x', 'o' }, ';', ts_repeat_move.repeat_last_move)
```
**Additional context**
This diverges from the way it works in core nvim and is inconsistent with `F`, `T`,
and visual selection behavior.

As far as i can tell, this is only reproducible when repeating moves from `builtin_f_expr` and 
`builtin_t_expr` on operation pending mode, not with `builtin_F_expr` and `builtin_T_expr`.

**Comment by kiyoon** [comment]
@phanen Could you open a PR for your fix? Your solution seems good to me.

**Comment by kiyoon** [comment]
Closed by #707

**Comment by kiyoon** [comment]
Fixed poorly with #707, reverted with #795, fixed again with #833.
